packages/flutter_tools/lib/src/commands/build_ios_framework.dart

```diff
@@ -0,0 +1,399 @@
+
+/// Produces a .framework for integration into a host iOS app. The .framework
+/// contains the Flutter engine and framework code as well as plugins. It can
+/// be integrated into plain Xcode projects without using or other package
+/// managers.
+class BuildIOSFrameworkCommand extends BuildSubCommand {
+  BuildIOSFrameworkCommand({this.aotBuilder, this.bundleBuilder}) {
+    usesTargetOption();
+    usesFlavorOption();
+    usesPubOption();
+    argParser
+      ..addFlag('debug',
+        negatable: true,
+        defaultsTo: true,
+        help: 'Whether to produce a framework for the debug build configuration. '
+              'By default, all build configurations are built.'
+      )
+      ..addFlag('profile',
+        negatable: true,
+        defaultsTo: true,
+        help: 'Whether to produce a framework for the profile build configuration. '
+              'By default, all build configurations are built.'
+      )
+      ..addFlag('release',
+        negatable: true,
+        defaultsTo: true,
+        help: 'Whether to produce a framework for the release build configuration. '
+              'By default, all build configurations are built.'
+      )
+      ..addFlag('universal',
+        help: 'Produce universal frameworks that include all valid architectures. '
+              'This is true by default.',
+        defaultsTo: true,
+        negatable: true
+      )
+      ..addFlag('xcframework',
+        help: 'Produce xcframeworks that include all valid architectures (Xcode 11 or later).',
+      )
+      ..addOption('output',
+        abbr: 'o',
+        valueHelp: 'path/to/directory/',
+        help: 'Location to write the frameworks.',
+      );
+  }
+
+  AotBuilder aotBuilder;
+  BundleBuilder bundleBuilder;
+
+  @override
+  final String name = 'ios-framework';
+
+  @override
+  final String description = 'Produces a .framework directory for a Flutter module '
+      'and its plugins for integration into existing, plain Xcode projects.\n'
+      'This can only be run on macOS hosts.';
+
+  @override
+  Future<Set<DevelopmentArtifact>> get requiredArtifacts async => const <DevelopmentArtifact>{
+    DevelopmentArtifact.universal,
+    DevelopmentArtifact.iOS,
+  };
+
+  @override
+  Future<void> validateCommand() async {
+    await super.validateCommand();
+    if (argResults['xcframework'] && xcode.majorVersion < 11) {
+      throwToolExit('--xcframework requires Xcode 11.');
+    }
+  }
+
+  @override
+  Future<FlutterCommandResult> runCommand() async {
+    final FlutterProject flutterProject = FlutterProject.current();
+    if (!flutterProject.isModule) {
+      throwToolExit('Building frameworks for iOS is only supported from a module.');
+    }
+
+    if (!platform.isMacOS) {
+      throwToolExit('Building frameworks for iOS is only supported on the Mac.');
+    }
+
+    if (!argResults['universal'] && !argResults['xcframework']) {
+      throwToolExit('--universal or --xcframework is required.');
+    }
```

Comment:
Should this also be moved to `validateCommand()`?
(I know I wrote this, but I didn't know validateCommand was a thing)

Reply:
There's no dartdoc at https://github.com/flutter/flutter/blob/ed82bb821cd7c62315ec5973e36855ca75f17350/packages/flutter_tools/lib/src/runner/flutter_command.dart#L627 lol.

@jonahwilliams what is the pattern supposed to be?

Reply:
though looking at the few commands which happens to override it, they seem to just put everything in it. I assume that's the intent?